include/sdf/parser.hh

```diff
@@ -388,6 +388,15 @@ namespace sdf
   SDFORMAT_VISIBLE
   bool checkJointParentChildLinkNames(const sdf::Root *_root);
 
+  /// \brief Check that all joints in contained models specify parent
+  /// and child link names that match the names of sibling links.
+  /// This checks recursively and should check the files exhaustively
+  /// rather than terminating early when the first error is found.
+  /// \param[in] _root SDF Root object to check recursively.
+  /// \param[out] _errors Detected errors will be appended to this variable.
+  SDFORMAT_VISIBLE
+  void checkJointParentChildLinkNames(const sdf::Root *_root, Errors &_errors);
```

Comment:
now that I think about it, we could change this function name since parent and child aren't just links anymore

* [ ] `checkJointParentChildNames`
* [ ] `checkJointParentChildFrameNames`